conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/feignimpl/EndpointNameHeaderEnrichmentContract.java

```diff
@@ -34,7 +39,23 @@ public EndpointNameHeaderEnrichmentContract(Contract delegate) {
     }
 
     @Override
-    protected void processMetadata(Class<?> _targetType, Method method, MethodMetadata metadata) {
+    protected void processMetadata(Class<?> targetType, Method method, MethodMetadata metadata) {
+        String httpMethod = metadata.template().method();
+        if (httpMethod == null) {
+            throw new SafeNullPointerException(
```

Comment:
Preconditions.checkNotNull?

Reply:
Opted against preconditions since the safe-arg values aren't direct references to existing data -- we don't need to compute those for each endpoint.